chainstate/src/detail/chainstateref/epoch_seal/epoch_seal_op.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{chain::ChainConfig, primitives::BlockHeight};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum EpochSealOp {
+    Seal,
+    Unseal,
+    None,
+}
+
+impl EpochSealOp {
+    pub fn new(
+        chain_config: &ChainConfig,
+        sealed_epoch_height: Option<BlockHeight>,
```

Comment:
I think it's safe to assume that if there's no seal, then it's at height zero. No need for this to be an Option. At the caller, it can be `unwrap_or(0)`.